src/main/java/org/wildfly/security/authz/jacc/ElytronPolicyProvider.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2013 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.authz.jacc;
+
+import org.wildfly.security.authz.jacc.action.GetPolicyAction;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+import javax.security.jacc.PolicyContext;
+import javax.security.jacc.PolicyContextException;
+import javax.security.jacc.WebResourcePermission;
+import java.security.CodeSource;
+import java.security.Permission;
+import java.security.PermissionCollection;
+import java.security.Permissions;
+import java.security.Policy;
+import java.security.Principal;
+import java.security.ProtectionDomain;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Set;
+
+import static java.security.AccessController.doPrivileged;
+import static org.wildfly.security._private.ElytronMessages.log;
+
+/**
+ * <p>A {@link java.security.Policy} implementation that knows how to process JACC permissions.</p>
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class ElytronPolicyProvider extends Policy {
```

Comment:
Accordingly with the specification, prolicy providers are actually implementations of java.security.Policy, which delegates non-jacc policy decisions to the corresponding default system Policy implementation class.

In PicketBox the equivalent class is named "DelegatingPolicy". I found "ElytronPolicyProvider" a better name and more in sync with the specification.

Also, the policy provider is not only about checking role-based permissions and performing mapping function. JACC defines other types of permissions (eg.: unchecked and excluded) as well.

Do you really want to change the name ?

Thanks.